ncm-opennebula/src/main/perl/opennebula.pm

```diff
@@ -306,12 +316,21 @@ sub change_oneadmin_passwd
     }
 }
 
+# Restart one service
+# after any conf change
+sub restart_opennebula_service {
+    my ($self) = @_;
+
+    my $srv = CAF::Service->new(['opennebula'], log => $self);
+    $srv->restart();
+}
+
 # Remove/add ONE resources
 # based on resource type
 sub manage_something
 {
-    my ($self, $one, $type, $resources) = @_;
-
+    my ($self, $one, $type, $resources, $untouchables) = @_;
+    my %untouch = map { $_ => 1 } @$untouchables;
```

Comment:
Actually, I'd use `Set::Scalar` instead of this. SPMA already depends on it, so you wouldn't be adding any extra dependencies.